src/Lykke.Service.Dash.Api/Controllers/BalancesController.cs

```diff
@@ -0,0 +1,101 @@
+        private readonly IBalanceRepository _balanceRepository;
+        private readonly IBalancePositiveRepository _balancePositiveRepository;
+
+        public BalancesController(ILog log, 
+            IDashService dashService,
+            IBalanceRepository balanceRepository,
+            IBalancePositiveRepository balancePositiveRepository)
+        {
+            _log = log;
+            _dashService = dashService;
+            _balanceRepository = balanceRepository;
+            _balancePositiveRepository = balancePositiveRepository;
+        }
+
+        [HttpGet]
+        public async Task<PaginationResponse<WalletBalanceContract>> Get([Required, FromQuery] int take, [FromQuery] string continuation)
+        {
+            var result = await _balancePositiveRepository.GetAsync(take, continuation);
+            
+            return PaginationResponse.From(
+                result.Continuation, 
+                result.Items.Select(f => f.ToWalletBalanceContract()).ToArray()
+            );
+       }
+
+        [HttpPost("{address}/observation")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        public async Task<IActionResult> AddToObservations([Required] string address)
+        {
+            if (string.IsNullOrEmpty(address))
+            {
+                return BadRequest(ErrorResponse.Create($"{nameof(address)} is null or empty"));
+            }
+
+            var validAddress = _dashService.GetBitcoinAddress(address) != null;
+            if (!validAddress)
+            {
+                return BadRequest(ErrorResponse.Create($"{nameof(address)} is not valid"));
+            }
+
+            var balance = await _balanceRepository.GetAsync(address);
+            if (balance != null)
+            {
+                return new StatusCodeResult(StatusCodes.Status409Conflict);
+            }
+
+            await _log.WriteInfoAsync(nameof(BalancesController), nameof(AddToObservations),
+                new { address = address }.ToJson(), "Add address to observations");
+
+            await _balanceRepository.AddAsync(address);
+
+            return Ok();
+        }
+
+        [HttpDelete("{address}/observation")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        public async Task<IActionResult> DeleteFromObservations([Required] string address)
+        {
+            if (string.IsNullOrEmpty(address))
+            {
+                return BadRequest(ErrorResponse.Create($"{nameof(address)} is null or empty"));
+            }
+
+            var balance = await _balanceRepository.GetAsync(address);
+            if (balance == null)
+            {
+                return new StatusCodeResult(StatusCodes.Status204NoContent);
+            }
+
+            await _log.WriteInfoAsync(nameof(BalancesController), nameof(DeleteFromObservations),
+                new { address = address }.ToJson(), "Delete address from observations");
+
+            await _balanceRepository.DeleteAsync(address);
```

Comment:
It case if execution fail after this line, data from _balancePositiveRepository will never be deleted